app/src/main/java/org/mozilla/focus/browser/BrowserNavigationOverlay.kt

```diff
@@ -367,6 +367,20 @@ class BrowserNavigationOverlay @JvmOverloads constructor(
         adapter.removeTile(tileId)
     }
 
+    /**
+     * Focus may be lost if all pinned items are removed via onContextItemSelected()
+     */
+    fun checkIfTilesFocusNeedRefresh() {
+        val adapter = tileContainer.adapter as HomeTileAdapter
+        if (adapter.itemCount == 0) {
+            if (pocketVideosContainer.isVisible) {
+                pocketVideoMegaTileView.requestFocus()
+            } else {
+                megaTileTryAgainButton.requestFocus()
+            }
+        }
+    }
```

Comment:
Optional nit: we could remove a bit of nesting here with something like the following:

```kotlin
val currentTiles = (tileContainer.adapter as RecyclerView.Adapter).itemCount
if (currentTiles == 0) return
val targetView = if (pocketVideosContainer.isVisible) {
    pocketVideoMegaTileView
  } else {
  megaTileTryAgainButton
}
targetView.requestFocus()
```

Up to you though, it's not a huge difference.